File: app/src/main/java/com/example/toothfairy/entity/DailyWearTime.java
package com.example.toothfairy.entity;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.sql.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;


@ToString
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class DailyWearTime {

    @Expose @SerializedName("id")
    private int id;

    @Expose @SerializedName("patientNum")
    private String patientNum;

    @Expose @SerializedName("wearDate")
    private Date wearDate;           // 착용 일자

    @Expose @SerializedName("totalWearTime")
    private Long totalWearTime;      // 하루 총 착용 시간 (milliseconds)
}

File: app/src/main/java/com/example/toothfairy/entity/DailyWearTime.kt
package com.example.toothfairy.entity

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.sql.Date

class DailyWearTime {
    @Expose
    @SerializedName("id")
    var id = 0

    @Expose
    @SerializedName("patientId")
    var patientId: String? = null

    @Expose
    @SerializedName("wearDate")
    var wearDate: Date? = null // 착용 일자

    @Expose
    @SerializedName("totalWearTime")
    var totalWearTime: Long? = null // 하루 총 착용 시간 (milliseconds)

    constructor(id: Int, patientId: String?, wearDate: Date?, totalWearTime: Long?): this() {
        this.id = id
        this.patientId = patientId
        this.wearDate = wearDate
        this.totalWearTime = totalWearTime
    }

    constructor() {}

    override fun toString(): String {
        return "DailyWearTime(id=$id, patientId=$patientId, wearDate=$wearDate, totalWearTime=$totalWearTime)"
    }
}